vars/scmNotify.groovy

```diff
@@ -0,0 +1,39 @@
+// vars/scmNotify.groovy
+
+ /* This provides a way of notifying the SCM such as GitLab/GitHub with
+  * the status of the build stage in progress.
+  *
+  * The scmNotifySystem must be provided as a shared
+  * groovy library local to the running Jenkins for this routine.
+  *
+  * If it is not provided this routine will not do anything.
+  */
+
+  /**
+   * scmNotify step method
+   *
+   * @param config Map of parameters passed
+   *
+   * See the githubNotify pipeline step for the parameters to pass.
+   */
+
+def call(Map config = [:]) {
+
+  def errtxt = 'Jenkins not configured to notify SCM repository of builds.'
+  if (env.DAOS_JENKINS_NOTIFY_STATUS == null) {
+    println errtxt
+    return
+  }
+  try {
+    scmNotifySystem(config)
```

Comment:
Is there any reason not to just pull vars/scmNotifySystem.groovy￼ up to here so that it adds the flexibility for somebody else to use gitlab instead of github?

Reply:
The gitlab notify in system-pipeline-lib will only work for our internal gitlab jobs currently.

I have not looked into what it would take to make it work with DAOS.

Reply:
Really?  It looks like the very generic equivalent of it's `githubNotify` counterpart.  What is specific to our internal gitlab?

Reply:
Now a wrapper to scmNotifyTrusted in trusted-pipeline-lib